pytorch_lightning/utilities/cli.py

```diff
@@ -275,3 +409,37 @@ def fit(self) -> None:
 
     def after_fit(self) -> None:
         """Implement to run some code after fit has finished"""
+
+
+def _global_add_class_path(class_type: Type, init_args: Dict[str, Any]) -> Dict[str, Any]:
+    return {
+        'class_path': class_type.__module__ + '.' + class_type.__name__,
+        'init_args': init_args,
+    }
+
+
+def _add_class_path_generator(class_type: Type) -> Callable[[Dict[str, Any]], Dict[str, Any]]:
+
+    def add_class_path(init_args: Dict[str, Any]) -> Dict[str, Any]:
+        return _global_add_class_path(class_type, init_args)
+
+    return add_class_path
```

Comment:
You could replace this with a `lambda` or `functools.partial`

Reply:
It is done like this because when an argument link is created which applies a function, in the help the link is shown as `target <-- function_name(source)`. If a lambda is used it would no look good in the help. I haven't used `functools.partial` so not sure about that one.

Reply:
I checked it. The current implementation shows:

```python
Linked arguments:
  model.optim1 <-- add_class_path(optim1) [applied on parse]
                        (type: <class 'dict'>)
  model.scheduler <-- add_class_path(lr_scheduler) [applied on parse]
                        (type: <class 'dict'>)
```

and replacing `add_class_path = partial(_global_add_class_path, class_type)` shows:

```python
Linked arguments:
  model.optim1 <-- functools.partial(<function _global_add_class_path at 0x12c4790d0>, <class 'torch.optim.adam.Adam'>)(optim1) [applied on parse]
                        (type: <class 'dict'>)
  model.scheduler <-- functools.partial(<function _global_add_class_path at 0x12c4790d0>, <class 'torch.optim.lr_scheduler.ExponentialLR'>)(lr_scheduler) [applied on parse]
                        (type: <class 'dict'>)
```

The latter looks scarier but has the benefit of showing the actual class used. What do you think?

We could subclass partial and set our custom repr

Reply:
Showing in the help something like `at 0x12c4790d0` is not great and can be confusing for a user. A custom repr could be but it would be like replacing the current generator with this subclass implementation. I think I don't have a preference.

Reply:
Let's go with the current impl, we can always do this minor improvement later